### PR TITLE
Refactor view item test

### PR DESCRIPTION
**Pull request by gestchild**
## Who is this for?
Maintenance
Relates to #10409

## What is it doing for them?
- Making the tests less flakey and faster
- Testing more stuff on mobile (we we unnecessarily excluding mobile from some of the tests, now we're not)
- Removed a couple of tests which were unnecessary/covered by the others

Running the old file on my local machine against a built app:

Before mobile:
Screenshot 2024-01-31 at 11 34 53

Before desktop:
Screenshot 2024-01-31 at 11 30 20

After mobile:
Screenshot 2024-01-31 at 11 38 00

After desktop:
Screenshot 2024-01-31 at 11 45 29

### CODE REVIEW

File: playwright/test/view-items.test.ts [hunk 1]
```diff
@@ -56,357 +27,250 @@ const test = base.extend({
...
-      expect(imageDownloadUrl).toBe(smallImageDownloadUrl);
-    }
-  );
-
-  multiVolumeTest('downloading the entire item', async ({ page }) => {
-    const fullItemDownloadElement = await page.waitForSelector(
-      fullItemDownload
-    );
-    const fullDownloadUrl = await fullItemDownloadElement.getAttribute('href');
+// test 14 very occasionally fails on desktop, so am allowing 1 retry
```
Comment (rcantin-w) [inline playwright/test/view-items.test.ts:38]: out of curiosity, which bit tends to fail?

File: playwright/test/view-items.test.ts [hunk 2]
```diff
@@ -56,357 +27,250 @@ const test = base.extend({
...
 
-test.describe("Scenario 9: A user wants to be able to search inside an item's text", () => {
-  test('the item should be searchable', async ({ page, context }) => {
-    await itemWithSearchAndStructures(context, page);
-    await safeWaitForNavigation(page);
-    if (isMobile(page)) {
-      await page.click('text="Show info"');
+test('(13) | The main viewer can be scrolled', async ({ page, context }) => {
+  await itemWithSearchAndStructures(context, page);
+  await expect(page.locator('[data-testid="main-viewer"] > div')).toBeVisible();
```
Comment (rcantin-w) [inline playwright/test/view-items.test.ts:195]: If you want, I _think_ you could do something like
`page.getByTestId('main-viewer').locator('> div')` but would need to be tested.

File: playwright/test/view-items.test.ts [hunk 3]
```diff
@@ -56,357 +27,250 @@ const test = base.extend({
...
+  await page.$eval(
```
Comment (rcantin-w) [inline playwright/test/view-items.test.ts:196]: Didn't know about this so had a quick look and it seems [to be discouraged](https://playwright.dev/docs/api/class-elementhandle#element-handle-eval-on-selector), but replaceable by `locator.evaluate()` https://playwright.dev/docs/api/class-locator#locator-evaluate

**Comment by rcantin-w** [comment]
I think we're good to kill `playwright/test/helpers/safeWaitForNavigation.ts`, which was the main flake-boi 🎉  if you wouldn't mind doing it as part of this PR?

**Comment by gestchild** [comment]
> I think we're good to kill `playwright/test/helpers/safeWaitForNavigation.ts`, which was the main flake-boi 🎉 if you wouldn't mind doing it as part of this PR?

Thanks, thought I'd done that

File: playwright/test/view-items.test.ts (hunk 3 above)
Comment (gestchild) [reply playwright/test/view-items.test.ts]: Good spot, thanks

**Pull request closed without merging**